src/graphnet/data/pipeline.py

```diff
@@ -0,0 +1,173 @@
+        """
+        self._pipeline_name = pipeline_name
+        self._device = device
+        self.n_workers =  n_workers
+        self._features = features
+        self._truth =  truth
+        self._batch_size = batch_size
+        self._outdir = outdir
+        self._module_dict = module_dict
+        self._retro_table_name = retro_table_name
+
+    def __call__(self, database, pulsemap) -> dict:
+        """Runs inference of each field in self._module_dict[target]['']
+
+        Args:
+            database (path): path to database with pulsemap and truth
+            pulsemap (str): name of pulsemaps
+        """
+        outdir = self._get_outdir(database)
+        if isinstance(self._device, str): # Because pytorch lightning insists on breaking pytorch cuda device naming scheme
+            device = int(self._device[-1])
+        if os.path.isdir(outdir) == False:
+            dataframes = []
+            for target in self._module_dict.keys():
+                dataloader = self._make_dataloader(database, pulsemap)
+                trainer = Trainer(gpus = [device])
+                model = torch.load(self._module_dict[target]['path'], map_location = 'cpu', pickle_module = dill)
+                model._gnn.to(self._device)
+                model._device = self._device
+                model.to(self._device)
+                model.eval()
+                model.inference()
+                results = get_predictions(trainer,model, dataloader, self._module_dict[target]['output_column_names'], ['event_no'])
+                dataframes.append(results.sort_values('event_no').reset_index(drop = True))
+            df = self._combine_outputs(dataframes)
+            self._make_pipeline_database(outdir,df, database)
+        else:
+            print(outdir)
+            print('WARNING - Pipeline named %s already exists! \n Please rename pipeline!'%self._pipeline_name)
+        return
+    def _load_modules(self, module_dict):
+        """Loads each module in _module_dict
+
+        Args:
+            module_dict (dict): dictionary of GNN ModuleList
+
+        Returns:
+            dict: module_dict with "loaded_module" field added.
+        """
+        for target in module_dict.keys():
+            model = torch.load(module_dict[target]['path'], map_location = 'cpu', pickle_module = dill)
+            model._gnn.to(self._device)
+            model._device = self._device
+            model.to(self._device)
+            model.eval()
+            module_dict[target]['loaded_module'] = model
+        return module_dict
+
+    def _make_dataloader(self, database, pulsemap):
+        dataloader = make_dataloader(db = database, pulsemaps = pulsemap, features = self._features, truth = self._truth, batch_size = self._batch_size, shuffle = False, selection = None, num_workers = self.n_workers, persistent_workers= False)
+        return dataloader
```

Comment:
A lot of the arguments to the `InSQLPipeline` class are only used as arguments to `make_dataloader` here. So I'm wondering if it would be simpler to drop this method and then, in the class constructor, just create an `self._dataloader = make_dataloader(...)` member variable rather than first setting `self._features`, `self._truth`, etc. as members and then only referencing them here?

Reply:
Not entirely sure what you mean. I did something - perhaps this is what you meant?